<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <!-- The above 3 meta tags *must* come first in the head; any other head content must come *after* these tags -->
    <title>rpy2: Python-R bridge</title>

    <script>
      (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
      })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

      ga('create', 'UA-8995033-4', 'auto');
      ga('send', 'pageview');
      
    </script>

    <!-- Bootstrap -->
    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.5.2/css/bootstrap.min.css" integrity="sha384-JcKb8q3iqJ61gNV9KGb8thSsNjpSL0n8PARn9HuZOnIxN0hoP+VmmDGMN5t9UJ0Z" crossorigin="anonymous">
    <link rel="stylesheet" href="css/custom.css">

  </head>
  <body>
    <nav class="navbar navbar-expand-lg fixed-top navbar-light bg-light">
      <!-- <div class="container"> -->
      <a class="navbar-brand" href="index.html"><img alt="logo" src="images/rpy2_logo2013.png" style="height: 2em"></a>
      <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
    <span class="navbar-toggler-icon"></span>
  </button>
      <div id="navbar" class="collapse navbar-collapse">
        <ul class="navbar-nav">
	  
	  
	  <li class="nav-item">
	    <a class="nav-link" href="rpy2_doc.html">Documentation </a>
	  </li>
	  
	  
	  <li class="nav-item">
	    <a class="nav-link" href="https://github.com/rpy2/rpy2-docker">Docker </a>
	  </li>
	  
	  
	  <li class="nav-item">
	    <a class="nav-link" href="https://github.com/rpy2/rpy2-mybinder">MyBinder </a>
	  </li>
	  
	  
	  <li class="nav-item">
	    <a class="nav-link" href="https://github.com/rpy2/rpy2-R6">R6 </a>
	  </li>
	  
	  
	  <li class="nav-item">
	    <a class="nav-link" href="https://github.com/rpy2/rpy2-Matrix">Matrix </a>
	  </li>
	  
        </ul>
      </div><!--/.nav-collapse -->
      <!-- </div> -->
    </nav>

    <div class="container">
      
      <div class="header">
        <!-- <nav> -->
        <!--   <ul class="nav nav-pills pull-right"> -->
        <!--     <li role="presentation" class="active"><a href="#">Home</a></li> -->
        <!--     <li role="presentation"><a href="#">About</a></li> -->
        <!--     <li role="presentation"><a href="#">Contact</a></li> -->
        <!--   </ul> -->
        <!-- </nav> -->
        <h3>rpy2 - R in Python</h3>
      </div>

      <div class="jumbotron">
	<p class="lead">
	  <img id="imglogo" src="images/rpy2_logo2013.png" alt="rpy2">
	  <a class="btn btn-lg btn-default" href="https://pypi.python.org/pypi/rpy2" role="button">
	    <img src="https://img.shields.io/pypi/v/rpy2.svg?style=flat-square"> on Pypi
	  </a>
	</p>
      </div>

      <div class="row">
        <div class="col-lg-6">
	  <div class="card">
	    <div class="card-header">About</div>
	    <div class="card-body">
	      <p class="card-text">
		rpy2 is an interface to <a href="http://www.r-project.org/">R</a>
		running embedded in a <a href="http://www.python.org">Python</a> process.
	      </p>
	      <p class="card-text">
		The project is mature, stable, <a href="https://rpy2.github.io/index_doc.html">
		  documented</a>, and widely used
		(<a href="http://rpy2.readthedocs.org/">older documentation</a>).
	      </p>
	    </div>
	  </div>
	  
	  <div class="card">
	    <div class="card-header">License</div>
	    <div class="card-body">
  	      <p class="card-text">The same license as R: GPLv2 or greater.
	      </p>
	    </div>
	  </div>

	  <div class="card">
	    <div class="card-header">Questions and issues</div>
	    <div class="card-body">
  	      <p class="card-text">
	    Consider having a look at the documentation.
	    Otherwise questions should preferably be asked on the
	    rpy mailing-list on SourceForge,
	    or on <a href="http://stackoverflow.com/search?q=rpy2">StackOverflow</a>.
	    Bugs, or wishes, can also go to the 
	    <a href="https://github.com/rpy2/rpy2/issues">Github page</a>
	    for the project.
	      </p>
	    </div>
	  </div>

	  <div class="card">
	    <div class="card-header">Source code</div>
	    <div class="card-body">
	      <h4 class="card-title">Repository</h4>
	      <p class="card-text">
		The source code is also in a public repository on
		<a href="https://github.com/rpy2/rpy2">Github</a>.
	      </p>
	    </div>
	  </div>	  
        </div>
	
        <div class="col-lg-6">
	  <div class="card">
	    <div class="card-header">Easy install</div>
	    <div class="card-body">
	      <h4 class="card-title">pip</h4>
	      <p class="card-text">
		Released source packages are available on 
		<a href="https://pypi.python.org/pypi/rpy2">PyPi</a>. Installing
		should be as easy<sup>*</sup> as<br/>
		<pre class="bg-dark text-white p-3"><code>pip install rpy2</code></pre>
		(*: except on Windows)
	      </p>
	      <h4 class="card-title">Docker</h4>
	      <p class="card-text">
		A Docker image is also available to "kick the tires"
		with the last build from <i>master</i>:
	      </p>
	      <div>
		a Jupyter notebook (on port 8888):
		<pre class="bg-dark text-white p-3"><code>docker run --rm -p 8888:8888 \
		    rpy2/jupyter-ubuntu:master-19.10</code></pre>
	      </div>
	      <div>
		an ipython console:
		<pre class="bg-dark text-white p-3"><code>docker run -it --rm \
		    rpy2/jupyter-ubuntu:master-19.10 ipython</code></pre>
	      </div>
	      <p class="card-text">
	    More information about Docker images can be found in
	    the repository <a href="https://github.com/rpy2/rpy2-docker">rpy2-docker</a>.
	      </p>
	    </div>
	  </div>
        </div>
	<div class="row">
          <div class="col-lg-12" style="padding-left: 30px; padding-right: 30px;">
	    <div class="card" style="margin-top: 20px;">
	      <div class="card-header">Documentation</div>
	      <div class="card-body">
		<p class="card-text">More information
		  can be found in the 
		  <a href="https://rpy2.github.io/index_doc.html">documentation</a>.
		</p>
		<p class="card-text">Additional resources, about rpy2 in particular or demonstrations
		  of polyglot data analysis using rpy2 to call R from Python,
		  are available (don't hesitate to notify us about
		  other resource, but avoid Python <i>vs</i> R trolls... unless
		  funny):
		  <ul class="list-group list-group-flush">
		    <li class="list-group-item">
		      <a href="http://lgautier.github.io/odsc-pdaibp-slides/">
			Polygot Data Analysis Is Being Pragmatic</a>
		      (Talk: <a href="https://www.odsc.com/boston">
			ODSC Boston 2016</a>)
		    </li>
		    <li class="list-group-item">
		      <a href="http://lgautier.github.io/jpd-pdapr-slides/">
			Polyglot Data Analysis Visually Demonstrated With Python And R</a>
		      (Talk: <a href="http://blog.jupyter.org/2016/02/29/jdboston16-speaker-announcement/">
			Jupyter Days Boston</a>)
		    </li>
		    <li class="list-group-item">
		      <a href="http://www.activestate.com/blog/2016/02/r-vs-python-false-dichotomy">
			R <i>vs</i> Python: a false dichotomy</a> (Blog: ActiveState)
		    </li>
		    <li class="list-group-item">
		      <a href="http://blog.revolutionanalytics.com/2016/01/pipelining-r-python.html">
			Pipelining R and Python in Notebooks
		      </a> (Blog: Revolutions Analytics / Microsoft)
		    </li>
		    <li class="list-group-item">
		      <a href="http://blog.yhat.com/posts/rpy2-combing-the-power-of-r-and-python.html">RPy2: Combining the Power of R + Python for Data Science</a> (Blog: yhat)
		    </li>
		  </ul>
		</p>
	      </div>
	    </div>
	  </div>
	</div>
	<div class="row">
          <div class="col-lg-12" style="padding-left: 30px; padding-right: 30px;">
	    <div class="card" style="margin-top: 20px;">
	      <div class="card-header">rpy2 in the wild</div>
	      <div class="card-body">
		<p class="card-text">
		  The package is used in a wide range of domains. For example:
		</p>
		<h4>2019</h4>
		<ul class="list-group list-group-flush">
		  <li class="list-group-item">
		   Diamantopoulos, Nikos, et al. "Engineering for a Science-Centric Experimentation Platform." arXiv preprint arXiv:1910.03878 (2019).
		  </li>
		  <li class="list-group-item">
		   Ponti, Alexandre, et al. "First-Line Selective Internal Radiation Therapy in Patient with Uveal Melanoma Liver Metastases." Journal of Nuclear Medicine (2019): jnumed-119.
		  </li>
		  <li class="list-group-item">
		   Levy, Joshua J., et al. "PyMethylProcess—convenient high-throughput preprocessing workflow for DNA methylation data." Bioinformatics (2019).
		  </li>
		  <li class="list-group-item">
		    Gruber, Michaela, et al. "Growth dynamics in naturally progressing chronic lymphocytic leukaemia." Nature (2019): 1.
		  </li>
		  <li class="list-group-item">
		    Dunnmon, Jared, et al. "Cross-Modal Data Programming Enables Rapid Medical Machine Learning." arXiv preprint arXiv:1903.11101 (2019).
		  </li>		  
		</ul>
		<h4>2018</h4>
		<ul class="list-group list-group-flush">
		  <li class="list-group-item">
		    Christophides, Damianos, et al. "A method for automatic selection of parameters in NTCP modelling." International Journal of Radiation Oncology• Biology• Physics (2018).
		  </li>
		  <li class="list-group-item">
		    Fadda, Giulia, et al. "MRI and laboratory features and the performance of international criteria in the diagnosis of multiple sclerosis in children and adolescents: a prospective cohort study." The Lancet Child & Adolescent Health (2018).
		  </li>
		  <li class="list-group-item">
		    Muschelli, John, et al. "Neuroconductor: an R platform for medical imaging analysis." Biostatistics (2018).
		  </li>
		  <li class="list-group-item">
		    Domingues, Rémi, et al. "A comparative evaluation of outlier detection algorithms: Experiments and analyses." Pattern Recognition 74 (2018): 406-421.
		  </li>
		  <li class="list-group-item">
		    Poirion, Olivier B., and Benedicte Lafay. "Neo-formation of chromosomes in bacteria." bioRxiv (2018): 264945.
		  </li>
		  <li class="list-group-item">
		  de Melo, Vinícius Veloso, and Wolfgang Banzhaf. "Automatic feature engineering for regression models with machine learning: An evolutionary computation and statistics hybrid." Information Sciences 430 (2018): 287-313.
		  </li>
		  <li class="list-group-item">
		    Giuliano, Christopher J., et al. "MELK expression correlates with tumor mitotic activity but is not required for cancer growth." eLife 7 (2018).
		  </li>
		  <li class="list-group-item">
		    De Vries, Stefan PW, et al. "Phylogenetic analyses and antimicrobial resistance profiles of Campylobacter spp. from diarrhoeal patients and chickens in Botswana." PloS one 13.3 (2018): e0194481.
		  </li>
		</ul>
		<h4>2017</h4>
		<ul class="list-group list-group-flush">
		  <li class="list-group-item">
		    Cherifi, Nadir, et al. "Automatic Inference of Energy Models for Peripheral Components in Embedded Systems." FiCloud 2017: The 5th International Conference on Future Internet of Things and Cloud. 2017.
		  </li>
		  <li class="list-group-item">
		    Lowe, Andrew John. "Language-agnostic data analysis workflows and reproducible research." (2017).
		  </li>
		  <li class="list-group-item">
		    Alexander, William M., et al. "multiplierz v2. 0: a Python‐based ecosystem for shared access and analysis of native mass spectrometry data." Proteomics (2017).
		  </li>
		  <li class="list-group-item">
		    Liu, Lingjie, et al. "An approach of identifying differential nucleosome regions in multiple samples." BMC genomics 18.1 (2017): 135.
		  </li>
		  <li class="list-group-item">
		    Olesen, Scott W., Claire Duvallet, and Eric J. Alm. "dbOTU3: A new implementation of distribution-based OTU calling." PloS one 12.5 (2017): e0176335.
		  </li>
		  <li class="list-group-item">
		    Peacock, Jacob, and Harish Sethu. "Which Leaflet is More Effective: A Reanalysis." (2017).
		  </li>
		  <li class="list-group-item">
		    Sharker, Yushuf, and Eben Kenah. "Estimation of the Household Secondary Attack Rate: Binomial Considered Harmful." arXiv preprint arXiv:1705.01135 (2017).
		  </li>
		  <li class="list-group-item">
		    Domingues, Rémi, et al. "A comparative evaluation of outlier detection algorithms: experiments and analyses." Pattern Recognition (2017).
		  </li>
		  <li class="list-group-item">
		    Zhao, Xuefang, Alexandra M. Weber, and Ryan E. Mills. "A recurrence based approach for validating structural variation using long-read sequencing technology." bioRxiv (2017): 105817.
		  </li>
		  <li class="list-group-item">
		    Laycock, Silas GT. "From blackbirds to black holes: Investigating capture-recapture methods for time domain astronomy." New Astronomy 54 (2017): 91-102.
		  </li>
		  <li class="list-group-item">
		    Chiera, Belinda A., and Małgorzata W. Korolkiewicz. "Visualizing Big Data: Everything Old Is New Again." Big Data Management. Springer International Publishing, 2017. 1-27.
		  </li>
		  <li class="list-group-item">
		    Amarante, Linda M., Marcelo S. Caetano, and Mark Laubach. "Medial frontal theta is entrained to rewarded actions." Journal of Neuroscience (2017): 1965-17.
		  </li>
		</ul>
		<h4>2016</h4>
		<ul class="list-group list-group-flush">
		  <li class="list-group-item">
		    Freer, Rosie, et al. "A protein homeostasis signature in healthy brains recapitulates tissue vulnerability to Alzheimer’s disease." Science Advances 2.8 (2016): e1600947.
		  </li>
		  <li class="list-group-item">
		    Haslwanter, Thomas. An Introduction to Statistics with Python. Springer, 2016.
		  </li>
		  <li class="list-group-item">
		    Ekmekci, Berk, Charles E. McAnany, and Cameron Mura. "An Introduction to Programming for Bioscientists: A Python-Based Primer." PLoS Comput Biol 12.6 (2016): e1004867.
		  </li>
		  <li class="list-group-item">
		    <a href="https://aws.amazon.com/blogs/compute/analyzing-genomics-data-at-scale-using-r-aws-lambda-and-amazon-api-gateway/">Analyzing genomics data at scale with R, AWS Lambda and Amazon API gateway</a> (AWS Compute Blog)
		  </li>
		  <li class="list-group-item">
		    Anaya, Jordan. "OncoLnc: Linking TCGA survival data to mRNAs, miRNAs, and lncRNAs." PeerJ PrePrints 4 (2016): e1780v1.
		  </li>
		  <li class="list-group-item">
		    Cleland, Edward John, et al. "The bacterial microbiome in chronic rhinosinusitis: Richness, diversity, postoperative changes, and patient outcomes." American journal of rhinology & allergy 30.1 (2016): 37-43.
		  </li>
		  <li class="list-group-item">
		    Vandenbulcke, Hélène, et al. "Alcohol intake increases the risk of hepatocellular carcinoma in patients with hepatitis C virus-related compensated cirrhosis: a prospective study." Journal of hepatology (2016).
		  </li>
		  <li class="list-group-item">
		    Jeong, Seongwoon, et al. "A NoSQL Data Management Infrastructure for Bridge Monitoring." (2016).
		  </li>
		  <li class="list-group-item">
		    Kenah, Eben, et al. "Molecular Infectious Disease Epidemiology: Survival Analysis and Algorithms Linking Phylogenies to Transmission Trees." PLoS Comput Biol 12.4 (2016): e1004869.
		  </li>
		  <li class="list-group-item">
		    Kittas, Aristotelis, et al. "Organizational principles of the Reactome human BioPAX model using graph theory methods." Journal of Complex Networks (2016): cnw003.
		  </li>
		</ul>
		<h4>2015</h4>
		<ul class="list-group list-group-flush">
		  <li class="list-group-item">
		    Zhang, Zhi-Min, et al. "Multiscale peak detection in wavelet space." Analyst 140.23 (2015): 7955-7964.
		  </li>
		  <li class="list-group-item">
		    Woon, Wei Lee, et al. "Changes in Occupational Skills-A Case Study Using Non-negative Matrix Factorization." Neural Information Processing. Springer International Publishing, 2015.
		  </li>
		  <li class="list-group-item">
		    Antao, Tiago. Bioinformatics with Python cookbook. Packt Publishing Ltd, 2015.
		  </li>
		  <li class="list-group-item">
		    Terna, Pietro. "Introducing the Swarm-Like Agent Protocol in Python (SLAPP)." Agent-based Models of the Economy. Palgrave Macmillan UK, 2015. 31-54.
		  </li>
		</ul>
		<h4>2014</h4>
		<ul class="list-group list-group-flush">
		  <li class="list-group-item">
		    Nakamura, Kunio, et al. "Correlation between brain volume change and T2 relaxation time induced by dehydration and rehydration: implications for monitoring atrophy in clinical studies." NeuroImage: Clinical 6 (2014): 166-170.
		  </li>
		  <li class="list-group-item">
		    Filosi, Michele, et al. "ReNette: a web-infrastructure for reproducible network analysis." bioRxiv (2014): 008433.
		  </li>
		  <li class="list-group-item">
		    Angeli, Nicole F., et al. A process to support species conservation planning and climate change readiness in protected areas. No. e492v2. PeerJ PrePrints, 2014.
		  </li>
		  <li class="list-group-item">Röst, Hannes L., et al. "pyOpenMS: A Python‐based interface to the OpenMS mass‐spectrometry algorithm library." Proteomics 14.1 (2014): 74-77.</li>
		  <li class="list-group-item">
		    Heavey, Cathal, et al. "Development of an open-source discrete event simulation cloud enabled platform." Proceedings of the 2014 Winter Simulation Conference. IEEE Press, 2014.
		  </li>
		  <li class="list-group-item">
		    De Melo, Vinícius Veloso. "Kaizen programming." Proceedings of the 2014 conference on Genetic and evolutionary computation. ACM, 2014.
		  </li>
		  <li class="list-group-item">
		    Onggo, B. S., et al. "TOWARDS AUTOMATED SIMULATION INPUT DATA." 8TH SIMULATION WORKSHOP. 2014.
		  </li>
		  <li class="list-group-item">
		    Via, Allegra, Kristian Rother, and Anna Tramontano. Managing Your Biological Data with Python. CRC Press, 2014.
		  </li>
		  <li class="list-group-item">
		    Brown, Robert A., Sridar Narayanan, and Douglas L. Arnold. "Imaging of repeated episodes of demyelination and remyelination in multiple sclerosis." NeuroImage: Clinical 6 (2014): 20-25.
		  </li>
		</ul>
		<h4>2013</h4>
		<ul class="list-group list-group-flush">
		  <li class="list-group-item">
		    Krishnan, Hari, et al. "Exploring Collaborative HPC Visualization Workflows using VisIt and Python." Proceedings of the 12th Python in Science Conference (SciPy 2013). 2013.
		  </li>
		  <li class="list-group-item">
		    Parks, Donovan H., et al. "GenGIS 2: Geospatial analysis of traditional and genetic biodiversity, with new gradient algorithms and an extensible plugin framework." PloS one 8.7 (2013): e69885.
		  </li>
		</ul>
		<h4>2012</h4>
		<ul class="list-group list-group-flush">
		  <li class="list-group-item">Harper, Marc, and Christopher J. Lee. "Genome-wide analysis of mutagenesis bias and context sensitivity of N-methyl-N′-nitro-N-nitrosoguanidine (NTG)." Mutation Research/Fundamental and Molecular Mechanisms of Mutagenesis 731.1 (2012): 64-67.</li>
		  <li class="list-group-item">Strickland, C. M., et al. "PyMCMC: A Python Package for Bayesian Estimation Using Markov Chain Monte Carlo." (2012).</li>
		  <li class="list-group-item">
		    Vitolo, Claudia, W. O. U. T. E. R. Buytaert, and D. Reusser. "Hydrological Models as Web Services: An Implementation Using OGC Standards." Proceedings of the 10th International Conference on Hydroinformatics, HIC, Hamburg, Germany. 2012.
		  </li>
		  <li class="list-group-item">
		    Talevich, Eric, et al. "Bio. Phylo: A unified toolkit for processing, analyzing and visualizing phylogenetic trees in Biopython." BMC bioinformatics 13.1 (2012): 1.
		  </li>
		  <li class="list-group-item">
		    Bagger, Frederik Otzen, et al. "HemaExplorer: a database of mRNA expression profiles in normal and malignant haematopoiesis." Nucleic acids research (2012): gks1021.
		  </li>
		</ul>
		<h4>2011</h4>
		<ul class="list-group list-group-flush">
		  <li class="list-group-item">
		    Buske, Orion J., et al. "Exploratory analysis of genomic segmentations with Segtools." BMC bioinformatics 12.1 (2011): 1.
		  </li>
		  <li class="list-group-item">Tolk, A., et al. "TOWARDS AUTOMATED SIMULATION INPUT DATA: AN OPEN SOURCE TOOL TO ENHANCE THE INPUT DATA PHASE IN DISCRETE EVENT SIMULATION."</li>
		  <li class="list-group-item">
		    Patton, Evan W., et al. "SemNExT: A Framework for Semantically Integrating and Exploring Numeric Analyses."
		  </li>
		  <li class="list-group-item">
		    Connolly, Daniel W., et al. "Integrating R efficiently to allow secure, interactive analysis within a clinical data warehouse." The 8th International R User Conference. Vanderbilt University.
		  </li>
		  <li class="list-group-item">
		    Masica, David L., and Rachel Karchin. "Correlation of somatic mutation and expression identifies genes important in human glioblastoma progression and survival." Cancer research 71.13 (2011): 4550-4561.
		  </li>
		  <li class="list-group-item">
		    Simone, James. "PoS (Lattice 2011) 048 Data analysis using the Gnu R system for statistical computation." (2011).
		  </li>
		  <li class="list-group-item">
		    Liu, Sanmin, et al. "HDX-analyzer: a novel package for statistical analysis of protein structure dynamics." BMC bioinformatics 12.1 (2011): 1.
		  </li>
		</ul>
		<h4>2009</h4>
		<ul class="list-group list-group-flush">
		  <li class="list-group-item">
		    Roseline, Bilina, and Steve Lawford. "Python for Unified Research in Econometrics and Statistics." Available at SSRN 1429822 (2009).
		  </li>
		  <li class="list-group-item">	
		    Wijffels, Jan. "Prediction and Fuzzy Logic at ThomasCook to automate price settings of last minute offers." (2009).
		  </li>
		  <li class="list-group-item">
		    McBeth, Rafe. "Computational investigation of biological dose-volume outcome predictors in 29 canine nasal tumor patients treated with stereotactic radiation therapy." (2007).
		  </li>
		</ul>
	      </div>
	    </div>
	  </div>
	</div>
      </div>


    </div><!-- /.container -->

    <script src="https://code.jquery.com/jquery-3.5.1.slim.min.js" integrity="sha384-DfXdz2htPH0lsSSs5nCTpuj/zy4C+OGpamoFVy38MVBnE+IbbVYUew+OrCXaRkfj" crossorigin="anonymous"></script>
    <script src="https://cdn.jsdelivr.net/npm/popper.js@1.16.1/dist/umd/popper.min.js" integrity="sha384-9/reFTGAW83EW2RDu2S0VKaIzap3H66lZH81PoYlFhbGU+6BZp6G7niu735Sk7lN" crossorigin="anonymous"></script>
    <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.5.2/js/bootstrap.min.js" integrity="sha384-B4gt1jrGC7Jh4AgTPSdUtOBvfO8shuf57BaghqFfPlYxofvL8/KUEfYiJOMMV+rV" crossorigin="anonymous"></script>    
  </body>
</html>
